lib/graphql/metrics.rb

```diff
@@ -51,8 +51,8 @@ def self.current_time_monotonic
       Process.clock_gettime(Process::CLOCK_MONOTONIC)
     end
 
-    def self.time(*args)
-      TimedResult.new(*args) { yield }
+    def self.time(offset_time = nil)
+      TimedResult.new(offset_time) { yield }
```

Comment:
```suggestion
    def self.time(offset_time = nil, &block)
      TimedResult.new(offset_time, &block)
```

Reply:
default argument:
```
Comparison:
Metrics.time_without_splat_and_block:  6667246.6 i/s
Metrics.time_without_splat:  6524671.1 i/s - same-ish: difference falls within error
        Metrics.time:  3756359.7 i/s - 1.77x  slower
```

explicit argument:
```
Comparison:
Metrics.time_without_splat:  6364099.5 i/s
Metrics.time_without_splat_and_block:  5018904.9 i/s - 1.27x  slower
        Metrics.time:  3401205.2 i/s - 1.87x  slower

```

Reply:
How so? It's avoiding an extra block. It's not a big deal, but it's one less call.

Reply:
I don't have the same results. Not sure which ruby you are using. But anyway, based on the implementation I'd be very surprised if `{ yield }` could be faster in any cases.